libraries/AP_HAL_SITL/SITL_State.h

```diff
@@ -106,23 +106,23 @@ class HALSITL::SITL_State {
     void _gps_write(const uint8_t *p, uint16_t size, uint8_t instance = 0);
     void _gps_send_ubx(uint8_t msgid, uint8_t *buf, uint16_t size, uint8_t instance);
     void _update_gps_ubx(const struct gps_data *d, uint8_t instance);
-    void _update_gps_mtk(const struct gps_data *d);
-    void _update_gps_mtk16(const struct gps_data *d);
-    void _update_gps_mtk19(const struct gps_data *d);
+    void _update_gps_mtk(const struct gps_data *d, uint8_t instance);
+    void _update_gps_mtk16(const struct gps_data *d, uint8_t instance);
+    void _update_gps_mtk19(const struct gps_data *d, uint8_t instance);
     uint16_t _gps_nmea_checksum(const char *s);
-    void _gps_nmea_printf(const char *fmt, ...);
-    void _update_gps_nmea(const struct gps_data *d);
-    void _sbp_send_message(uint16_t msg_type, uint16_t sender_id, uint8_t len, uint8_t *payload);
-    void _update_gps_sbp(const struct gps_data *d);
+    void _gps_nmea_printf(uint8_t instance, const char *fmt, ...);
+    void _update_gps_nmea(const struct gps_data *d, uint8_t instance);
+    void _sbp_send_message(uint16_t msg_type, uint16_t sender_id, uint8_t len, uint8_t *payload, uint8_t instance);
+    void _update_gps_sbp(const struct gps_data *d, uint8_t instance);
     void _update_gps_file(const struct gps_data *d);
```

Comment:
I think this isn't used much, if at all, but can we change it to have instances too? It would read from a gps2.dat file instead.